File: Cmdlets/GetThreeLetterWindowsLanguageNameCommand.vb
<System.Management.Automation.Cmdlet("Get", "ThreeLetterWindowsLanguageName")> _
Public Class GetThreeLetterWindowsLanguageNameCommand
    Inherits System.Management.Automation.Cmdlet

    Protected Overrides Sub BeginProcessing()

        Dim Cultures As New System.Collections.Generic.List(Of System.Tuple(Of String, System.Globalization.CultureInfo))

        For Each CC In System.Globalization.CultureInfo.GetCultures(System.Globalization.CultureTypes.AllCultures)
            Cultures.Add(New System.Tuple(Of String, System.Globalization.CultureInfo)(CC.ThreeLetterWindowsLanguageName, CC))
        Next

        Dim TLW = Cultures.ToLookup(Function(T) T.Item1, Function(T) T.Item2)

        Dim TLWLN As New System.Collections.Generic.List(Of ThreeLetterWindowsLanguageName)

        For Each Lang In TLW

            Dim TL As New ThreeLetterWindowsLanguageName

            TL.Name = Lang.Key

            For Each TC In Lang
                TL.Cultures.Add(TC)
            Next

            TLWLN.Add(TL)
        Next

        Me.WriteObject(TLWLN, True)
    End Sub
End Class

File: Cmdlets.Core/Commands/GetThreeLetterWindowsLanguageNameCommand.cs
using System.Linq;
using TIKSN.Cmdlets.Common.Models;

namespace TIKSN.Cmdlets.Core.Commands
{
    [System.Management.Automation.Cmdlet("Get", "ThreeLetterWindowsLanguageName")]
    public class GetThreeLetterWindowsLanguageNameCommand : System.Management.Automation.Cmdlet
    {
        protected override void BeginProcessing()
        {
            System.Collections.Generic.List<System.Tuple<string, System.Globalization.CultureInfo>> Cultures = new System.Collections.Generic.List<System.Tuple<string, System.Globalization.CultureInfo>>();

            foreach (var CC in System.Globalization.CultureInfo.GetCultures(System.Globalization.CultureTypes.AllCultures))
            {
                Cultures.Add(new System.Tuple<string, System.Globalization.CultureInfo>(CC.ThreeLetterWindowsLanguageName, CC));
            }

            var TLW = Cultures.ToLookup(T => T.Item1, T => T.Item2);

            System.Collections.Generic.List<ThreeLetterWindowsLanguageName> TLWLN = new System.Collections.Generic.List<ThreeLetterWindowsLanguageName>();

            foreach (var Lang in TLW)
            {
                ThreeLetterWindowsLanguageName TL = new ThreeLetterWindowsLanguageName();

                TL.Name = Lang.Key;

                foreach (var TC in Lang)
                {
                    TL.Cultures.Add(TC);
                }

                TLWLN.Add(TL);
            }

            this.WriteObject(TLWLN, true);
        }
    }
}